VMEncryption/main/test/test_check_util.py

```diff
@@ -523,3 +528,76 @@ def test_check_kv_name(self):
+        with self.assertRaises(Exception):
+           self.cutil.preInitializationCheck(logger=self.logger,imdsStoredResults=mockimdsStoredResults,iMDSUtil=mockimdsUtil)
+        mockimdsUtil.get_security_type_IMDS.return_value="ad"
+        with self.assertRaises(Exception):
+            self.cutil.preInitializationCheck(logger=self.logger,imdsStoredResults=mockimdsStoredResults,iMDSUtil=mockimdsUtil)
+        mockimdsUtil.get_security_type_IMDS.return_value="TrustedLaunch2"
+        with self.assertRaises(Exception):
+            self.cutil.preInitializationCheck(logger=self.logger,imdsStoredResults=mockimdsStoredResults,iMDSUtil=mockimdsUtil)      
+
+    def test_preinitializationcheckInValidInputsRegistryCheck(self):
+        from Common import CommonVariables
+        import os
+        mockencryption_environment = Mock()
+        mockimdsUtil = Mock()
+        current_path = os.getcwd()
+        mockencryption_environment.imds_stored_results_file_path = os.path.join(current_path,'imds_stored_results')
+        imdsStoredResults = IMDSStoredResults(logger=self.logger,encryption_environment=mockencryption_environment)
+        #non valid inputs
+        mockimdsUtil.get_security_type_IMDS.return_value="ConfidentialVM1"
+        with self.assertRaises(Exception):
+           self.cutil.preInitializationCheck(logger=self.logger,imdsStoredResults=imdsStoredResults,iMDSUtil=mockimdsUtil)
+        self.assertEqual(imdsStoredResults.config_file_exists(),False)
+        self.assertEqual(imdsStoredResults.security_type,None)
+
+    def test_preinitializationcheckRegistry(self):
+        from Common import CommonVariables
+        import os
+        mockencryption_environment = Mock()
+        mockimdsUtil = Mock()
+        current_path = os.getcwd()
+        mockencryption_environment.imds_stored_results_file_path = os.path.join(current_path,'imds_stored_results')
+        imdsStoredResults = IMDSStoredResults(logger=self.logger,encryption_environment=mockencryption_environment)
+        #check if file exist, remove it.
+        if(imdsStoredResults.config_file_exists()):
+            os.remove(mockencryption_environment.imds_stored_results_file_path)
+        #confidential VM
+        mockimdsUtil.get_security_type_IMDS.return_value=CommonVariables.ConfidentialVM
+        with self.assertRaises(Exception):
+            self.cutil.preInitializationCheck(logger=self.logger,imdsStoredResults=imdsStoredResults,iMDSUtil=mockimdsUtil)
+        self.assertEqual(imdsStoredResults.get_security_type(),CommonVariables.ConfidentialVM)
+        #2nd call to pre initialization, reading data from stored file
+        mockimdsUtil.get_security_type_IMDS.return_value=""
+        with self.assertRaises(Exception):
+            self.cutil.preInitializationCheck(logger=self.logger,imdsStoredResults=imdsStoredResults,iMDSUtil=mockimdsUtil)
+        self.assertEqual(imdsStoredResults.get_security_type(),CommonVariables.ConfidentialVM)
```

Comment:
Since the return value is mocked to "", I'd expect the assertion is for type Standard?

Reply:
IMDS doesn't has security type information for Standard VM, and returning empty string, thus If we have IMDS replay as "" we consider it Standard VM.